src/mgr/DaemonState.h

```diff
@@ -261,9 +261,13 @@ class DaemonStateIndex
   template<typename Callback, typename...Args>
   auto with_daemons_by_server(Callback&& cb, Args&&... args) const ->
     decltype(cb(by_server, std::forward<Args>(args)...)) {
-    std::shared_lock l{lock};
-    
-    return std::forward<Callback>(cb)(by_server, std::forward<Args>(args)...);
+    decltype(by_server) by_server_copy;
```

Comment:
Making this `const` would exclude at least hiding local modifications to `by_server_copy`. `std::add_const<...>` could be useful here.

Reply:
Something like this?

`    const decltype(by_server) by_server_copy = [&] {
        std::shared_lock l{lock};
        return by_server;
    }();
`

I know less than nothing about std::add_const but, from what I'm reading, it's more for template metaprogramming types of scenarios and would be unnecessarily complex here for no gain? Am I reading it correctly?

Reply:
If compilers are happy with `const` here, I will be happy too :-).